Code/GraphMol/Wrap/rdchem.cpp

```diff
@@ -47,6 +52,86 @@ void wrap_EditableMol();
 void wrap_monomerinfo();
 void wrap_resmolsupplier();
 
+struct PySysErrWrite : std::ostream, std::streambuf
+{
+  std::string prefix;
+  
+  PySysErrWrite(const std::string &prefix) :
+      std::ostream(this), prefix(prefix) {}
+
+  int overflow(int c) { write(c); return 0;}
+  
+#ifdef RDK_THREADSAFE_SSS
+  void write(char c) { // enable thread safe logging
+    static boost::thread_specific_ptr< std::string > buffer;
+    if( !buffer.get() ) {
+      buffer.reset( new std::string );
+    }
+    (*buffer.get()) += c;
+    if (c == '\n') {
+      // Python IO is not thread safe, so grab the GIL
+      PyGILState_STATE gstate;
+      gstate = PyGILState_Ensure();
+      PySys_WriteStderr("%s", (prefix + (*buffer.get())).c_str());
+      PyGILState_Release(gstate);
+      buffer->clear();
+    }
+  }
+  
+#else
+  std::string buffer; // unlimited! flushes in endl
+  void write(char c) {
+    buffer += c;
+    if (c == '\n') {
+      PySys_WriteStderr("%s", (prefix + buffer).c_str());
+      buffer.clear();
+    }
+  }
+#endif
+
+  
+};
+
+void RDLogError(const std::string &msg) {
+  BOOST_LOG(rdErrorLog) << msg.c_str() << std::endl;
+}
+
+void RDLogInfo(const std::string &msg) {
+  BOOST_LOG(rdInfoLog) << msg.c_str() << std::endl;
+}
+
+void RDLogDebug(const std::string &msg) {
+  BOOST_LOG(rdDebugLog) << msg.c_str() << std::endl;
+}
+
+void RDLogWarning(const std::string &msg) {
+  BOOST_LOG(rdWarningLog) << msg.c_str() << std::endl;
+}
+
+#ifdef RDK_TEST_MULTITHREADED
+// As much as I hate adding test code to the python module
+//  we need to expose this to python somehow with the GIL
```

Comment:
I don't really care about Debug and Info, Warning is where the parsing errors go as they just return None but don't tell you what happened.